### PR TITLE
Move win32 files to their own platform folder

### PR DESCRIPTION
**Pull request by razzeee**
Like already done for android and some other platforms, I moved the win32 folder from:
xbmc/win32
to
xbmc/platform/win32

It's building fine for windows. But I'm not sure why it fails with cmake, http://jenkins.kodi.tv/view/Windows/job/WIN-32/9431/console maybe @fetzerch can have a look.

### CODE REVIEW

**Comment by Razzeee** [comment]
Rebased.

**Comment by afedchin** [comment]
looks good

**Comment by MartijnKaijser** [comment]
jenkins build this please 
Should we merge this now cause it's not like we changed actual code and it would give others time to adjust their PRs in case needed?

File: xbmc/threads/platform/win/ThreadImpl.cpp [hunk 1]
```diff
@@ -21,7 +21,7 @@
 #include <windows.h>
 #include <process.h>
 #include "threads/platform/win/Win32Exception.h"
-#include "../../win32/WIN32Util.h"
+#include "../../platform/win32/WIN32Util.h"
```
Comment (Paxxi) [inline xbmc/threads/platform/win/ThreadImpl.cpp]: Remove the../.. Includes should be from root and not relative

**Comment by Paxxi** [comment]
I think the decision should be up to @fetzerch as he is the one most impacted. I have no objections to merging other than that.

**Comment by fetzerch** [comment]
Change looks fine, I haven't tried it though. From what I can see it won't conflict much with my upcoming work, I can handle the rebase.
Reading the comments I'm not sure now if the build was fixed or not, but I kicked off another build: http://jenkins.kodi.tv/job/WIN-32/9504/

File: xbmc/threads/platform/win/ThreadImpl.cpp (hunk 1 above)
Comment (Razzeee) [inline xbmc/threads/platform/win/ThreadImpl.cpp]: Will do

**Comment by Razzeee** [comment]
@fetzerch 
thank you, would have done the same, maybe with a rebase first, but let's see if it builds

**Comment by MartijnKaijser** [comment]
jenkins build this please

**Comment by fetzerch** [comment]
@Razzeee: Please pick https://github.com/fetzerch/xbmc/commit/c82fad01f8a428528eea702016b2c2cb6eeb95df.

**Comment by Razzeee** [comment]
Done, if http://jenkins.kodi.tv/view/Windows/job/WIN-32/9511/ builds fine, we should be good

**Comment by MartijnKaijser** [comment]
let's get it in now to not hold up @fetzerch 
There's no code changes which could alter the working.

**Comment by Razzeee** [comment]
jenkins build and merge

**Pull request merged**